pkg/api/auth_middleware.go

```diff
@@ -148,11 +160,107 @@ func enhanceWithFriendlyName(user *model.User, friendlyName string) *model.User
 	return user
 }
 
+// userFromSAML returns a user from an existing SAML session.
+// If the user doesn't exist on the lakeFS side, it is created.
+// This function does not make any calls to an external provider.
+func userFromSAML(ctx context.Context, logger logging.Logger, authService auth.Service, authSession *sessions.Session, cookieAuthConfig *config.CookieAuthVerification) (*model.User, error) {
```

Comment:
Missing tests for this critical function.

Reply:
100%, we also don't have tests for `userFromOIDC`, will open a followup task to add tests for both!